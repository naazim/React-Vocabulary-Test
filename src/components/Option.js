import React from 'react';

const Option = (props) => (
  <div className="option">
      <span className="option__text">{props.nativeWord}</span>
      <span className="option__text">{props.foreignWord}</span>
    <button
      className="button button--icon"
      onClick={(e) => {
        props.handleDeleteWord(props.count - 1);
      }}
    >
      âœ–
      </button>
  </div>
);

export default Option;
