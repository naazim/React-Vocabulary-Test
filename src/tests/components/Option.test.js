import React from 'react';
import { shallow } from 'enzyme';
import Option from '../../components/Option';

test('should show the option panel with words recieved', () => {
    const wrapper = shallow(<Option nativeWord="morning" foreignWord="morgan" />);
    expect(wrapper.find('.option__text').at(0).text()).toEqual('morning');
    expect(wrapper.find('.option__text').at(1).text()).toEqual('morgan');
    expect(wrapper).toMatchSnapshot();
});

test('should delete the word pair on delete click', () => {
    const handleDeleteWordSpy = jest.fn();
    const wrapper = shallow(<Option nativeWord="morning" foreignWord="morgan" handleDeleteWord={handleDeleteWordSpy} />);
    wrapper.find('.button--icon').simulate('click');
    expect(handleDeleteWordSpy).toHaveBeenCalled();
});