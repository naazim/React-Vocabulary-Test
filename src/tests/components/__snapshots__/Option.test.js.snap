// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show the option panel with words recieved 1`] = `
<div
  className="option"
>
  <span
    className="option__text"
  >
    morning
  </span>
  <span
    className="option__text"
  >
    morgan
  </span>
  <button
    className="button button--icon"
    onClick={[Function]}
  >
    âœ–
  </button>
</div>
`;
